validation/management/commands/writetranscriptions.py

```diff
@@ -64,40 +82,60 @@ def handle(self, *args, **options):
             if not speaker:
                 continue
 
-            # Create necessary directories if they do not exist
-            speaker_dir = audio_dir / speaker
-            speaker_audio_dir = audio_dir / speaker / "wav"
-            speaker_persephone_dir = audio_dir / speaker / "label"
-            speaker_s4a_dir = audio_dir / speaker / "s4a"
-
-            for _dir in [
-                speaker_dir,
-                speaker_audio_dir,
-                speaker_persephone_dir,
-                speaker_s4a_dir,
-            ]:
-                _dir.mkdir(exist_ok=True)
-
-            # Copy the audio file
-            from_dir = audio_file
-            to_dir = audio_dir / speaker / "wav" / audio_filename
-            shutil.copyfile(from_dir, to_dir)
+            self.make_directories(training_dir, speaker)
+            self.copy_audio_file(training_dir, audio_file, speaker)
 
             # Treat the transcription for Persephone and save it
             persephone_filename = audio_id + ".txt"
-            persephone_path = audio_dir / speaker / "label" / persephone_filename
+            if "auto-validated" in query:
```

Comment:
This is really brittle, imo! I think it's okay for now (no changes needed), but the way I'd do it is that `write_transcriptions` would _not_ take in `query`, but instead take in something like the "kind" (maybe there's a better variable name):

```python
def write_transcriptions(self, audio_dir, kind: Literal["all", "auto-validated"]):
    if kind == "all":
        query = ... # the first query
    elif kind == "auto-validated":
        query = ... # the second query
```

Then this line would be:

```suggestion
            if kind == "auto-validated":
```

Less brittle. But I don't think it _really_ matters in this context.

Reply:
No no, I like this better